### ISSUE TITLE
The max and min value change on the fly

### ISSUE THREAD

**Issue by datnguyen293** [rzajac/angularjs-slider#184]
Hi, I'm writing an application which has search function with rzajac angularjs-slider as price filter with max value. When user submits a search request, the browser will pull the result from the server each second and append the results to view. While pulling the result, the price max value will be changed and I need the angularjs-slider changed according to the new value. But unfortunately, the slider was initialized at the page load and NOT reflect when max value is changed.

My setup as below:
```html
<rzslider rz-slider-model="filters.priceSlider" rz-slider-options="priceSliderOptions"></rzslider>
```

```javascript
$scope.filters.maxPrice = 1000;
$scope.priceSliderOptions = {
	showSelectionBar: true,
	step: 1,
	floor: 0,
	ceil: $scope.filters.maxPrice,
	translate: function (value) {
		return $filter('number')(value, 0) + 'usd';
	}
};
```

When new price is pulled from the server, I update the `$scope.filters.maxPrice` variable but the slider is stay still without any changes. Please help!

Thanks and best regards,
Dat

**Comment by ValentinH** [comment]
This is probably because when you set `ceil: $scope.filters.maxPrice`, the value is copied to ceil not passed as a reference (because it's a number, not an object). Thus, when you modify `$scope.filters.maxPrice`, you need to update `$scope.priceSliderOptions.ceil` as well.

I know that this is a drawback with the new version compared to the old `rz-slider-ceil` attribute, but I don't really know how to achieve the same behaviour. I had different ideas but none of them were acceptable to me...

On your side, if you don't want to do it manually everytime your `maxPrice`changes, you can create a `$watch` on `$scope.filters.maxPrice` and update the ceil. Creating a `$watch` might be consider as a bad practice but this is basically what was done internally by Angular when using `rz-slider-ceil`.

**Comment by datnguyen293** [comment]
You've pointed me to the right problem. I've resolved my problem and the slider is now working perfectly.
The problem was the `$scope.filters.maxPrice` is declared on parent controller then when I modified it in child controller it will not modify the parent variable.
I resolved my problem by moving the `$scope.filters` object to a `factory` then I can modified its properties in any controllers - this is the way I share variables between controllers as well.

Thank you again for excellent support!
